src/main/java/de/tum/in/www1/artemis/repository/GradingScaleRepository.java

```diff
@@ -63,7 +76,18 @@ default GradingScale findByCourseIdOrElseThrow(Long courseId) {
      */
     @NotNull
     default GradingScale findByExamIdOrElseThrow(Long examId) {
-        return findByExamId(examId).orElseThrow(() -> new EntityNotFoundException("Grading scale with exam ID " + examId + " doesn't exist"));
+        try {
+            return findByExamId(examId).orElseThrow(() -> new EntityNotFoundException("Grading scale with exam ID " + examId + " doesn't exist"));
+        }
+        catch (IncorrectResultSizeDataAccessException exception) {
+            List<GradingScale> gradingScales = findAllByExamId(examId);
```

Comment:
Can you please add some documentation why this code is necessary.
Do you have a server integration tests that checks if it works?

There is also code duplication here, can you please extract a method for the duplicated code?